docs/basic-features/script.md

```diff
@@ -13,102 +24,141 @@ description: Next.js helps you optimize loading third-party scripts with the bui
+/>
 ```
 
-## Examples
+Examples of scripts that should be loaded as soon as possible with this strategy include:
 
-### Loading Polyfills
+- Bot detectors
+- Cookie consent managers
 
-```js
-import Script from 'next/script'
+#### afterInteractive
 
-export default function Home() {
-  return (
-    <>
-      <Script
-        src="https://polyfill.io/v3/polyfill.min.js?features=IntersectionObserverEntry%2CIntersectionObserver"
-        strategy="beforeInteractive"
-      />
-    </>
-  )
-}
+Scripts that use the `afterInteractive` strategy are injected client-side and will run after Next.js hydrates the page. This strategy should be used for scripts that do not need to load as soon as possible and can be fetched and executed immediately after the page is interactive.
+
+```jsx
+<Script
+  strategy="afterInteractive"
+  dangerouslySetInnerHTML={{
+    __html: `
+    (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+    new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+    'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+    })(window,document,'script','dataLayer', 'GTM-XXXXXX');
+  `,
+  }}
+/>;
 ```
 
-### Lazy-Loading
+Examples of scripts that are good candidates to load immediately after the page becomes interactive include:
 
-```js
-import Script from 'next/script'
+- Tag managers
+- Analytics
 
-export default function Home() {
-  return (
-    <>
-      <Script
-        src="https://connect.facebook.net/en_US/sdk.js"
-        strategy="lazyOnload"
-      />
-    </>
-  )
-}
+#### lazyOnLoad
+
+Scripts that use the `lazyOnLoad` strategy are loaded late after all resources have been fetched and during idle time. This strategy should be used for background or low priority scripts that do not need to load before or immediately after a page becomes interactive. 
+
+```jsx
+<Script
+  src="https://connect.facebook.net/en_US/sdk.js"
+  strategy="lazyOnload"
+/>
 ```
 
+Examples of scripts that do not need to load immediately and can be lazy-loaded include:
+
+- Chat support plugins
+- Social media widgets
+
+### Inline Scripts
+
+Inline scripts, or scripts not loaded from an external file, are also supported by the Script component. They can be written by placing the JavaScript within curly braces:
+
+```jsx
+<Script id="show-banner" strategy="lazyOnload">
+  {`document.getElementById('banner').classList.remove('hidden')`}
+</Script>
+```
+
+Or by using the `dangerouslySetInnerHTML` property:
+
+```jsx
+<Script
+  id="show-banner"
+  dangerouslySetInnerHTML={{
+    __html: `document.getElementById('banner').classList.remove('hidden')`
+  }}
+/>
+```
+
+There are two limitations to be aware of when using the Script component for inline scripts:
+
+- Only the `afterInteractive` and `lazyOnLoad` strategies can be used. **The `beforeInteractive` loading strategy cannot be used with inline scripts.**
```

Comment:
```suggestion
- Only the `afterInteractive` and `lazyOnload` strategies can be used. **The `beforeInteractive` loading strategy cannot be used with inline scripts.**
```

Reply:
should this mention *why* the `beforeInteractive` strategy does not allow inline scripts?

Reply:
Fixed, and explained in a little detail why `beforeInteractive` cannot be used